ruby/ext/google/protobuf_c/message.c

```diff
@@ -242,9 +242,14 @@ static int extract_method_call(VALUE method_name, MessageHeader* self,
   // Method calls like 'has_foo?' are not allowed if field "foo" does not have
   // a hasbit (e.g. repeated fields or non-message type fields for proto3
   // syntax).
-  if (accessor_type == METHOD_PRESENCE && test_f != NULL &&
-      !upb_fielddef_haspresence(test_f)) {
-    return METHOD_UNKNOWN;
+  if (accessor_type == METHOD_PRESENCE && test_f != NULL) {
+    if (!upb_fielddef_haspresence(test_f)) return METHOD_UNKNOWN;
+
+    // TODO(haberman): remove this case, allow for proto3 oneofs.
```

Comment:
Do we previously support this?

Reply:
No, we have never supported this. The unit tests were failing until I added this check, see:

https://github.com/protocolbuffers/protobuf/blob/18950451c7bfb1497f1deaf1c3868e79e332d80a/ruby/tests/basic.rb#L45-L50